helix-core/src/main/java/org/apache/helix/monitoring/mbeans/ClusterStatusMonitor.java

```diff
@@ -898,26 +896,19 @@ private void unregisterPerInstanceResources(
       throws MalformedObjectNameException {
     synchronized (_perInstanceResourceMonitorMap) {
       for (PerInstanceResourceMonitor.BeanName beanName : beanNames) {
-        unregister(getObjectName(
-            getPerInstanceResourceBeanName(beanName.instanceName(), beanName.resourceName())));
+        _perInstanceResourceMonitorMap.get(beanName).unregister();
```

Comment:
Better to check null to avoid NPE.

Reply:
Unregister already has a null check like this. Should be enough, right?
public static void unregister(ObjectName objectName) {
    if (objectName != null && _beanServer.isRegistered(objectName)) {
      try {
..................

Reply:
_perInstanceResourceMonitorMap.get(beanName) could be null?